File: katas/cough-syrup-gcd-solv-java/src/main/code/org/codingkata/unit/MyKata.java
package org.codingkata.unit;

import org.codingkata.unit.api.BaseKataSolution;

public class MyKata extends BaseKataSolution {

    @Override
    public int calcGCD(int a, int b) {

        if (b == 0) {
            return a;
        }
        return calcGCD(b, a % b);
    }

}

File: katas/teleprompter-v2-solv-jython/src/main/code/org/codingkata/unit/MyKata.py
from org.codingkata.unit.api import BaseKataSolution;

class MyKata(BaseKataSolution):
    def translate(self, text, mapping):
        # My jython has no 're', else I'd use that...
        mapping = dict((str(key), str(value))
                       for key, value in mapping)
        bleeps = dict((str(key), "BLEEP")
                       for key, value in mapping.items())
        maps = (mapping, bleeps)

        text = text.split('"')

        for i, passage in enumerate(text):
            for key, value in maps[i % 2].items():
                passage = passage.replace(key, value)
            text[i] = passage

        return '"'.join(text)
